### ISSUE TITLE
Add multiplexer blocks for simple selections on enumerations

### ISSUE THREAD

**Issue by comonoidial** [wandernauta/viskell#255]
For small choice expressions using pattern matching or guards in a full case constructs is a bit too much.
An example is expression like:    abs x = if x < 0 then negate x else x
In hardware diagrams the multiplexer is a common block to define choice with little visual overhead.
So it would be useful to add a multiplexer variation that works on bools, and maybe all enumeration types.

**Comment by wandernauta** [comment]
Would the multiplexer block be like the conditional block from #97?

**Comment by comonoidial** [comment]
Yes, a generalization of that with the look of the ones used in hardware diagrams.

**Comment by wandernauta** [comment]
You mean like this?

![](https://i.imgur.com/yVu7Krn.png)

That would have anchors on the side, which I'm not sure really fits our line-drawing method.

**Comment by comonoidial** [comment]
I would rotate this so that the x's are on top, but where to put the selection one is an open question.

**Comment by wandernauta** [comment]
Something like this might work.

![screenshot](https://cloud.githubusercontent.com/assets/126832/11089871/b844239e-886d-11e5-91ed-0809d14d0c44.png)

There would be anchors, of course - the left one would take a `(Bounded a, Enum a)`, then there would be an x anchor for every x between `minBound` and `maxBound`, in that order.
